### PR TITLE
DEV-8199: Remove column in identify_cash_items where remove == True

### PR DESCRIPTION
**Pull request by thomasdotred**
# Pull Request Checklist

- [ ] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [ ] Tests pass

# Description of the PR

Describe the code changes for the reviewers, explain the solution you have provided and how it fixes the issue

### CODE REVIEW

File: tests/unit/cocoon/test_utilities.py [hunk 1]
```diff
@@ -1674,10 +1674,14 @@ def test_identify_cash_items(self, _, cash_flag, remove_cash_items, ground_truth
         dataframe, mappings_test = identify_cash_items(
             dataframe, mappings, file_type, remove_cash_items
         )
-
-        self.assertEqual(
-            ground_truth, list(dataframe["__currency_identifier_for_LUSID"])
-        )
+        if not remove_cash_items:
```
Comment (rizwansaeed) [inline tests/unit/cocoon/test_utilities.py:1677]: you should avoid having if logic in tests, these should be split into 2 tests

File: tests/unit/cocoon/test_utilities.py [hunk 2]
```diff
@@ -1587,24 +1587,44 @@ def test_identify_cash_items_failed(
...
-                    "implicit": "internal_currency",
-                },
-                False,
-                ["GBP_EXP", "GBP_EXP", "USD_EXP", "USD_IMP"],
-            ),
+                ["GBP_EXP", "GBP_EXP", "USD_EXP", "USD_EXP", None],
+            )
+        ]
+    )
+    def test_identify_cash_items_remove_false(self, _, cash_flag, remove_cash_items, ground_truth):
```
Comment (rizwansaeed) [inline tests/unit/cocoon/test_utilities.py:1594]: couple of things:
1) its not clear from the name what this test is meant to be doing, i think it should be renamed to something like `test_remove_cash_items_with_identify_cash_items`
2) i think `ground_truth` is your expected values, rename to something like `expected_values`
3) `remove_cash_items` doesn't look like it needs to be passed in and can be set in the call - its `False` for all

File: tests/unit/cocoon/test_utilities.py [hunk 3]
```diff
@@ -1587,24 +1587,44 @@ def test_identify_cash_items_failed(
...
+            "Currency"
+        ] = "__currency_identifier_for_LUSID"
+
+        dataframe = pd.DataFrame(data)
+
+        dataframe, mappings_test = identify_cash_items(
+            dataframe, mappings, file_type, remove_cash_items
+        )
+
+        self.assertEqual(
```
Comment (rizwansaeed) [inline tests/unit/cocoon/test_utilities.py:1620]: might be better to use `assertListEqual`

File: tests/unit/cocoon/test_utilities.py [hunk 4]
```diff
@@ -1645,41 +1665,38 @@ def test_identify_cash_items_failed(
                     "implicit": "internal_currency",
                 },
                 True,
-                [],
+                ["inst5"],
             ),
         ]
     )
-    def test_identify_cash_items(self, _, cash_flag, remove_cash_items, ground_truth):
+    def test_identify_cash_items_remove_true(self, _, cash_flag, remove_cash_items, ground_truth):
```
Comment (rizwansaeed) [inline tests/unit/cocoon/test_utilities.py:1672]: as above for naming and removing `remove_cash_items` parameter

File: tests/unit/cocoon/test_utilities.py (hunk 2 above)
Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py:1594]: 3. ✔️

Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py:1594]: 2. ✔️

Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py:1594]: 1, ✔️ :
`test_identify_cash_items_with_remove_cash_items()`
`test_identify_cash_items_without_remove()`

File: tests/unit/cocoon/test_utilities.py (hunk 4 above)
Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py:1672]: `test_identify_cash_items_without_remove()` as above

File: tests/unit/cocoon/test_utilities.py (hunk 3 above)
Comment (thomasdotred) [reply tests/unit/cocoon/test_utilities.py:1620]: Made change, also used assertDictEquals

File: tests/unit/cocoon/test_utilities.py (hunk 2 above)
Comment (rizwansaeed) [reply tests/unit/cocoon/test_utilities.py]: there are still some "ground truths" in there, can you rename to "expected values" or something equivalent

**Comment by rizwansaeed** [comment]
i think the sonar build has picked up a potential bug, can you fix

**Comment by thomasdotred** [comment]
> i think the sonar build has picked up a potential bug, can you fix

The code smells are where I used a list comprehension, but don't use the iterating variable. I have replaces `i` with `_` and sonar cloud has stopped complaining about it.

**Comment by thomasdotred** [comment]
The code duplications are two assertions that are the  same across two of the tests. I think that abstracting these tests will reduce readability of the tests so I have left it out

**Pull request merged**